### PR TITLE
Mono Repo

### PR DESCRIPTION
**Pull request by byronz**
testing
- [x] restructure unit test
- [x] resturcture integration test
- [x] optimize the ci config
- [x] linting and correcting the errors

packaging 
- [ ] adapt the build and release workflow with dev branch and new mono repo
- [ ] add bundle assets only in master branch, dev branch should get ignored

general repo TODOs
- [x] set up dev branch 
- [x] update social review image
- [x] add codeowners file
- [x] merge and restyle the readme.md https://github.com/plotly/dash/pull/713
- [x] adapt the issue template  in https://github.com/plotly/dash/pull/717
- [x] rework on the contribute.md  @alexcjohnson  draft is in the PR, please review
- [x] move the PR checklist into a [github PR template](https://help.github.com/en/articles/creating-a-pull-request-template-for-your-repository), fixes in https://github.com/plotly/dash/pull/721

- [x] add deprecation on dash-renderer and pytest-dash

### CODE REVIEW

**Comment by byronz** [comment]
@alexcjohnson @Marc-Andre-Rivet  

the latest commit should fix all integration errors, most of the code part refactoring is done at this stage. would like to have some feedback. I will continue with the rest of bureautic tasks.

**Comment by byronz** [comment]
Screen Shot 2019-05-10 at 3 57 48 PM

feel free to comment on the emoji choice

File: dash-renderer/index.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    <body>
+        <div id="react-entry-point"></div>
+    </body>
+
+    <script type="text/javascript">
+
+    </script>
+
+    <script type="text/javascript" src="build/bundle.js"></script>
+</html>
```
Comment (alexcjohnson) [inline dash-renderer/index.html]: what is this file for?

File: tests/integration/dash_assets/test_assets.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+class TestAssets(IntegrationTests):
+
+    def setUp(self):
+        def wait_for_element_by_id(id_):
+            wait_for(lambda: None is not invincible(
+                lambda: self.driver.find_element_by_id(id_)
+            ))
+            return self.driver.find_element_by_id(id_)
+        self.wait_for_element_by_id = wait_for_element_by_id
```
Comment (alexcjohnson) [inline tests/integration/dash_assets/test_assets.py]: Why the closure pattern here? Can't this just be part of `IntegrationTests`?

File: tests/integration/test_integration.py [hunk 3]
```diff
@@ -87,7 +88,7 @@ def update_output(value):
             len('hello world')
         )
 
-        assert_clean_console(self)
+        self.assertTrue(self.is_console_clean())
```
Comment (alexcjohnson) [inline tests/integration/test_integration.py]: Seems like a kind of funny pattern, and can't report what is in the console if it fails. Do we ever test that the console is *not* clean? Perhaps better as just `self.assert_clean_console()` or something like that?

File: tests/integration/dash_assets/test_assets.py (hunk 2 above)
Comment (byronz) [reply tests/integration/dash_assets/test_assets.py]: I didn't check the details, it was a simple 🚚  

In general all these wait_for needs a better refactoring

File: tests/integration/test_integration.py (hunk 3 above)
Comment (byronz) [reply tests/integration/test_integration.py]: personally, I am not a big fan of the assertTrue, assertEqual pattern. I like the pytest way, just using the native `assert`.  this one can be 
```python
assert self.is_console_clean(), "the browser console should contain no js error"
```
 I will clean up these in a separate PR

File: dash-renderer/index.html (hunk 1 above)
Comment (byronz) [reply dash-renderer/index.html]: moved together from dash-renderer, untouched chris index.html since 2017 :)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -2,133 +2,116 @@
...
-3. Comment in the PR with the prerelease version
-4. Update the top-level comment to include info about how to install, a summary of the changes, and a simple example.
-    * This makes it easier for a community member to come in and try it out. As more folks review, it's harder to find the installation instructions deep in the PR
-    * Keep this top-level comment updated with installation instructions (e.g. the `pip install` command)
-5. Make a post in the [Dash Community Forum][]
-    * Title it `":mega: Announcement! New <Your Feature> - Feedback Welcome"`
-    * In the description, link to the PR and any relevant issue(s)
-    * Pin the topic so that it appears at the top of the forum for two weeks
-
-## [Checklists](http://rs.io/unreasonable-effectiveness-of-checklists/)
```
Comment (byronz) [inline CONTRIBUTING.md]: @alexcjohnson  I will move the checklist section into a template file https://help.github.com/en/articles/creating-a-pull-request-template-for-your-repository

**Pull request merged**